[RFC] API: Use `ps -o comm` in nvim_get_proc()

lucc opened the pull request:
The ucomm argument to -o seems to be less portable than the comm version. See
commit message for links to docs.

This was noticed with the alpine docker image like so
```
docker run -t alpine sh -c 'apk add neovim && nvim --headless --cmd "echo nvim_get_proc(1)" --cmd q'
```

lucc committed:
API: Use `ps -o comm` in nvim_get_proc()

- The POSIX version of ps(1) only specifies "comm" for the "-o" option
  but not "ucomm".  See
  http://pubs.opengroup.org/onlinepubs/9699919799/utilities/ps.html
- On Linux (with the procps-ng package) "ucomm" is an alias for "comm".
  See https://gitlab.com/procps-ng/procps
- OpenBSD also has "ucomm" as an alias for "comm" (with the extra note
  "Name to be used for accounting."). See https://man.openbsd.org/ps
- FreeBSD describes "ucomm" as "Name to be used for accounting." but
  does not say that it should be an alias for "comm". See
  https://www.freebsd.org/cgi/man.cgi?query=ps

justinmk commented:
@lucc thanks for the thorough explanation in the commit message.

Test failed on travis macOS (which is probably why I chose `ucomm`):

```
[1m[35m[  FAILED  ][0m[0m [36m...is/build/neovim/neovim/test/functional/api/proc_spec.lua[0m @ [36m58[0m: [1mapi nvim_get_proc returns process info[0m
...is/build/neovim/neovim/test/functional/api/proc_spec.lua:61: Expected objects to be the same.
Passed in:
(string) '/Users/travis/build/neovim/neovim/build/bin/nvim'
Expected:
(string) 'nvim'
```

I guess the test could be relaxed to check the last 4 chars of the process name:

    eq('nvim', string.sub(name, string.len(name) - 3))

but probably it's better for `nvim_get_proc` to chop of the absolute path. Later we might want to include a new field `fullname` in the `nvim_get_proc` return value, which contains the full command string.

lucc commented:
I was trying to find official man pages for macOS some time ago for another project, without luck. Does anybody here know a webpage with the official man pages for macOS programs?

lucc committed:
API: Remove path prefix from command name in nvim_get_proc()

On macOS the output from `ps -o comm` might contain the full path of the
executable.  The `ucomm` would be the basename only but is less portable
(see previous commit).

justinmk commented:
@lucc Thanks! 

> I have trouble understanding where and why the quickbuild failed. How does one find the actual error there?

Search for `not ok` and `Terminated`.

However, the quickbuild linux/macOS envs currently are broken (mostly because of https://github.com/neovim/neovim/issues/8102 ) and we're ignoring them.
